### DISCUSSION TITLE
Providing build output in Azure

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3051] [Q&A] [answered]
I would like to show build results when failed. Has anyone found a good way to show this in Azure so the developers know it's build error?

Is there a helper extension out there that helps on this?

**Comment by devlead** [comment]
@nick5454 can you clarify what you mean about outputting to Azure? Are you talking about Azure DevOps or something like Azure Monitor / AppInsights?

**Reply by nick5454** [reply]
This is in Azure Devops. The developers check in code that has errors and I'm looking for a way to show the errors reported when calling MSBuild inside my cake script. 

I'd like to add a build error tab inside the pipeline results, so they can see the errors. But. i'm hoping there is a cake package set up for this already.

**Reply by devlead** [reply]
Ah then the [Cake.BuildSystems.Module](https://github.com/agc93/Cake.BuildSystems.Module) is probably what you're looking for.

You can add that to your script using the [module pre processor directive](https://cakebuild.net/docs/writing-builds/preprocessor-directives#module-directive)

```cake
#module nuget:?package=Cake.BuildSystems.Module&version=0.3.2
```

Depending on your version of Cake you might need to bootstrap script first using [--boostrap](https://cakebuild.net/docs/running-builds/runners/dotnet-tool#switches) switch.

I.e.
```
dotnet cake --bootsrap build.cake
dotnet cake build.cake
```

**Comment by pascalberger** [comment, marked as answer]
[Cake.Issues addins](https://cakeissues.net/) might be of interest to you. The [MsBuild provider](https://cakeissues.net/docs/issue-providers/msbuild/examples) allows to parse an MsBuild logfile for errors and warnings which you then can pass to a report provider to [Create an HTML report](https://cakeissues.net/docs/report-formats/generic/templates/htmldxdatagrid) or a [SARIF file](https://cakeissues.net/docs/report-formats/sarif/examples) which you then can show as a build tab using the [Azure DevOps SARIF SAST Scans Tab extension](https://marketplace.visualstudio.com/items?itemName=sariftools.scans). 
 
See [here](https://dev.azure.com/cake-contrib/Cake.Issues.Reporting.Sarif/_build/results?buildId=5495&view=sariftools.sarif-viewer-build-tab.sariftools.sarif-viewer-build-tab) for an example of the second approach.

Cake.Issues addins would allow you to also further process the errors and warnings, e.g. by [writing them as comments to pull requests](https://cakeissues.net/docs/pull-request-systems/azure-devops/features).

**Comment by nick5454** [comment]
Awesome! Thanks for the advice

**Reply by nick5454** [reply]
@pascalberger That is a great tool and really great UI debugging.

Is there a way to customize the tab name from "Scans" to something else like "Build Info"?

**Reply by pascalberger** [reply]
@nick5454 I'm not aware of any. But you might want to post the question in the [Q&A section of the extension](https://marketplace.visualstudio.com/items?itemName=sariftools.scans&ssr=false#qna).

For this extension I unfortunately also don't know if source code is available and where. The [predecessor extension](https://marketplace.visualstudio.com/items?itemName=sariftools.sarif-viewer-build-tab) had its source code available and there the name of the tab [was hardcoded](https://github.com/microsoft/sarif-azuredevops-extension/blob/4dfb7fb15642e221816f9f2bc9b68c89199e375a/vss-extension.prod.json#L13). But the new extension doesn't seem to be releases from this repo anymore as last commit doesn't match last publish date of the extension.

**Reply by nick5454** [reply]
@pascalberger Thanks so much for the tips
